src/main/java/eu/deltasource/internship/hotel/service/BookingService.java

```diff
@@ -13,9 +25,174 @@
+        }
+        throw new ItemNotFoundException("Invalid Id");
+    }
+
+    public void updateBookingByRoomId(int bookingId, Booking newBooking) {
+        if (!bookingValidation(newBooking)) {
+            throw new FailedInitializationException("Invalid Booking!");
+        }
+
+        int searchedRoomId = roomService.getRoomById(newBooking.getRoomId()).getRoomId();
+        int currentBookingRoomID = bookingRepository.findById(bookingId).getRoomId();
+
+        if (bookingRepository.existsById(bookingId) && (currentBookingRoomID == searchedRoomId)) {
+            deleteById(bookingId);
+            bookingRepository.save(newBooking);
+        }
+        throw new ItemNotFoundException("Invalid Id");
+    }
+
+    /**
+     * Checks if the new date overlaps the room's booking
+     *
+     * @param from        starting date
+     * @param to          ending date
+     * @param roomBooking room's booking
+     * @return
+     */
+    public boolean checkOverlapping(LocalDate from, LocalDate to, Booking roomBooking) {
+
+        if (from.isAfter(roomBooking.getFrom()) && to.isAfter(roomBooking.getTo())) {
+            return false;
+        }
+
+        Period period = Period.between(from, to);
+        int totalDays = period.getDays();
+
+        if (from.plusDays(totalDays).isBefore(roomBooking.getFrom())
+                || from.plusDays(totalDays).isEqual(roomBooking.getFrom())) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Finds all bookings
+     *
+     * @return list of all bookings
+     */
+    public List<Booking> findAll() {
+        return bookingRepository.findAll();
+    }
+
+    /**
+     * Creates new booking
+     *
+     * @param newBooking the new booking
+     */
+    public void save(Booking newBooking) {
+        if (!bookingValidation(newBooking)) {
+            throw new FailedInitializationException("Invalid booking!");
+        }
+        bookingRepository.save(newBooking);
+    }
+
+    /**
+     * Creates array of bookings
+     *
+     * @param bookings array of bookings
+     */
+    public void saveAll(Booking... bookings) {
+        if (!bookingValidation(bookings)) {
+            throw new FailedInitializationException("Invalid bookings!");
+        }
+        bookingRepository.saveAll(bookings);
+    }
+
+    private boolean dateValidation(LocalDate from, LocalDate to) {
+        if (from.isAfter(to) || from.equals(to) || from == null || to == null)
+            return false;
+        return true;
+    }
+
+    private boolean bookingOverlappingValidation(LocalDate from, LocalDate to, Booking booking) {
+        int roomID = booking.getRoomId();
+
+        for (Booking book : bookingRepository.findAll()) {
+            if (roomID == book.getRoomId() && checkOverlapping(from, to, book)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private boolean bookingValidation(Booking... booking) {
+        if (booking == null)
```

Comment:
Let's check the booking itself, instead the array. Also let's validate the Booking fields are present